packages/typespec-ts/test/modularUnit/modelsGenerator.spec.ts

```diff
@@ -1320,4 +1316,466 @@ describe("inheritance & polymorphism", () => {
+        protobuf: "text/vnd.ms.protobuf",
+      }
+      
+      op get(
+        @header("test-header") testHeader: SchemaContentTypeValues,
+        @body body: string,
+      ): { @header("test-header") testHeader: SchemaContentTypeValues };
+      `;
+      const schemaOutput = await emitModularModelsFromTypeSpec(
+        tspDefinition,
+        false,
+        true
+      );
+      assert.ok(schemaOutput);
+      await assertEqualContent(
+        schemaOutput?.getFullText()!,
+        `
+        /** Alias for SchemaContentTypeValues */
+        export type SchemaContentTypeValues =
+          | "application/json; serialization=Avro"
+          | "application/json; serialization=json"
+          | "text/plain; charset=utf-8"
+          | "text/vnd.ms.protobuf";
+        `
+      );
+    });
+
+    it("anonymous union with string literals being used in regular headers", async () => {
+      const tspDefinition = `
+      import "@typespec/http";
+      import "@typespec/rest";
+
+      @service({
+        title: "Widget Service",
+      })
+      namespace DemoService;
+      
+      using TypeSpec.Http;
+      using TypeSpec.Rest;
+      
+      op get(
+        @header("test-header") testHeader: "A" | "B",
+        @body body: string,
+      ): { @header("test-header") testHeader: "A" | "B" };
+      `;
+      const schemaOutput = await emitModularModelsFromTypeSpec(
+        tspDefinition,
+        false,
+        true
+      );
+      assert.isUndefined(schemaOutput);
+
+      const paramOutput = await emitModularOperationsFromTypeSpec(
+        tspDefinition,
+        true,
+        false,
+        false,
+        true
+      );
+      assert.ok(paramOutput);
+      assert.strictEqual(paramOutput?.length, 1);
+      const operationContent = paramOutput?.[0]?.getFullText()!;
+      await assertEqualContent(
+        operationContent,
+        `
+        import { DemoServiceContext as Client } from "../rest/index.js";
+        import {
+          StreamableMethod,
+          operationOptionsToRequestParameters,
+          createRestError,
+        } from "@azure-rest/core-client";
+        export function _getSend(
+          context: Client,
+          testHeader: "A" | "B",
+          body: string,
+          options: GetOptions = { requestOptions: {} },
+        ): StreamableMethod<Get204Response> {
+          return context
+            .path("/")
+            .post({
+              ...operationOptionsToRequestParameters(options),
+              headers: { "test-header": testHeader },
+            });
+        }
+        export async function _getDeserialize(result: Get204Response): Promise<void> {
+          if (result.status !== "204") {
+            throw createRestError(result);
+          }
+          return;
+        }
+        export async function get(
+          context: Client,
+          testHeader: "A" | "B",
+          body: string,
+          options: GetOptions = { requestOptions: {} },
+        ): Promise<void> {
```

Comment:
In this case, we actually have response headers returne, but we just returned void in the modular layer, I wonder if we should rethink about this a little ? @xirzec @joheredi

Reply:
here's the typespec
```
      op get(
        @header("test-header") testHeader: "A" | "B",
        @body body: string,
      ): { @header("test-header") testHeader: "A" | "B" };
```

Reply:
create an issue https://github.com/Azure/autorest.typescript/issues/2202 to track this.